src/store/authSlice.ts

```diff
@@ -0,0 +1,124 @@
+import { createSlice, createAsyncThunk, PayloadAction } from "@reduxjs/toolkit";
+import axios from 'axios';
+import { AxiosError } from 'axios';
+
+interface signUp{
+    FirstName: string,
+    LastName: string,
+    Email: string,
+    Password: string,
+}
+
+interface signIn{
+    Email: string,
+    Password: string
+}
+
+type KnownErr = {
+    message: string | null;
+}
+
+export const signup = createAsyncThunk<UserInfo, signUp, {rejectValue:KnownErr}>('auth/signup', async({FirstName, LastName, Email, Password}, {rejectWithValue}) =>{
+    try{
+        const res = await axios.post(`${process.env.REACT_APP_API}/signup`, {FirstName, LastName, Email, Password})
+        return res.data
+    }catch (err){
+        const error: AxiosError<KnownErr> = err as any;
+        if (!error.response) {
+          throw err;
+        }
+        return rejectWithValue(error.response.data);
+      }
+})
+
+export const signin = createAsyncThunk<UserInfo, signIn, {rejectValue:KnownErr}>('auth/signin', async({Email, Password}, {rejectWithValue}) =>{
+    try{
+        const res = await axios.post(`${process.env.REACT_APP_API}/signin`, {Email, Password})
+        return res.data
+    }catch (err){
+        const error: AxiosError<KnownErr> = err as any;
+        if (!error.response) {
+          throw err;
+        }
+        return rejectWithValue(error.response.data);
+      }
+})
+
+
+
+interface reducer {
+    user: string;
+    userId: string;
+}
```

Comment:
Not sure the name of the interface represent its content or what it is used for, think of a better name for it